### PR TITLE
Remaining tests from spec#Expressions

### PR DESCRIPTION
**Pull request by Dveim**

### CODE REVIEW

File: docs/quasiquotes.md [hunk 1]
```diff
@@ -33,20 +33,20 @@
...
- Annotate          | `q"$expr: ..@$expr"`
+ Annotate          | `q"$expr: ..@$annots"`
  Tuple             | `q"(..$exprs)"`
  Block             | `q"{ ..$stats }"`
  If                | `q"if ($expr) $expr else $expr"`
  Match             | `q"$expr match { ..case $cases }"`
- Try Catch Cases   | `q"try $expr catch { ..case $cases } finally $expropt"`
- Try Catch Expr    | `q"try $expr catch $expr finally $expropt" `
+ Try Catch Cases   | `q"try $expr catch { ..case $cases } finally expropt"`
+ Try Catch Expr    | `q"try $expr catch $expr finally expropt"`
```
Comment (xeno-by) [inline docs/quasiquotes.md]: Why are the dollars gone?

File: docs/quasiquotes.md [hunk 2]
```diff
@@ -250,6 +250,7 @@
  meta.Enumerator          | `$enumerator` | `enumerator`
  meta.Member              | `$member`     | `q`
  meta.Mod                 | `$mod`        | `mod`
+ meta.Mod.Annot           | `$annot`      | `mod`
```
Comment (xeno-by) [inline docs/quasiquotes.md]: This interface needs to be added to the public API

File: foundation/src/main/scala/org/scalameta/ast/ast.scala [hunk 3]
```diff
@@ -250,7 +250,7 @@ class AstMacros(val c: Context) {
       // step 10: finish codegen for Quasi
       qstats1 += q"def pt: _root_.java.lang.Class[_] = _root_.org.scalameta.runtime.arrayClass(_root_.scala.Predef.classOf[$iname], this.rank)"
       if (is("Term.Block.Quasi") || is("Type.Bounds.Quasi") || is("Ctor.Primary.Quasi") ||
-          is("Import.Clause.Quasi") || is("Lit.String.Quasi") || is("Mod.Annot.Quasi")) {
+          is("Import.Clause.Quasi") || is("Lit.String.Quasi")) {
```
Comment (xeno-by) [inline foundation/src/main/scala/org/scalameta/ast/ast.scala]: Is this change still necessary&

File: docs/quasiquotes.md (hunk 1 above)
Comment (Dveim) [inline docs/quasiquotes.md]: Typo, fixed.

File: foundation/src/main/scala/org/scalameta/ast/ast.scala (hunk 3 above)
Comment (Dveim) [inline foundation/src/main/scala/org/scalameta/ast/ast.scala]: Yes. Otherwise `[error] scala.NotImplementedError: implementation restriction: dangerous quasi, see the sources of @ast for more information`

File: docs/quasiquotes.md (hunk 2 above)
Comment (Dveim) [inline docs/quasiquotes.md]: Done.
